pkg/vere/main.c

```diff
@@ -1960,55 +1962,163 @@ _cw_pack(c3_i argc, c3_c* argv[])
   u3m_stop();
 }
 
-/* _cw_play(): replay events.
+/* _cw_play_slog(): print during replay.
+*/
+static void
+_cw_play_slog(u3_noun hod)
+{
+  u3_pier_tank(0, 0, u3k(u3t(hod)));
+  u3z(hod);
+}
+
+/* _cw_play_exit(): exit immediately.
+*/
+static void
+_cw_play_exit(c3_i int_i)
+{
+  //  explicit fprintf to avoid allocation in u3l_log
+  //
+  fprintf(stderr, "\r\n[received keyboard stop signal, exiting]\r\n");
+  raise(SIGINT);
+}
+
+/* _cw_play(): replay events, but better.
 */
 static void
 _cw_play(c3_i argc, c3_c* argv[])
 {
-  c3_i ch_i, lid_i;
+  c3_i lid_i, ch_i;
   c3_w arg_w;
+  c3_o ful_o = c3n;
+  c3_o mel_o = c3n;
+  c3_d eve_d = 0;
+  c3_d sap_d = 0;
 
-  static const c3_c usage_c[] = "error: invalid usage, expected "
-                                "`urbit play [--replay-to <event_num> "
-                                "| --batch-size <event_cnt>] <pier>`";
   static struct option lop_u[] = {
-    { "batch-size", required_argument, NULL, 'b' },
-    { "replay-to",  required_argument, NULL, 'n' },
-    { NULL,         0,                 NULL, 0   }
+    { "loom",      required_argument, NULL, c3__loom },
+    { "no-demand", no_argument,       NULL, 6 },
+    { "auto-meld", no_argument,       NULL, 7 },
+    { "full",      required_argument, NULL, 'f' },
+    { "replay-to", no_argument,       NULL, 'n' },
+    { "snap-at",   no_argument,       NULL, 's' },
+    { NULL, 0, NULL, 0 }
   };
 
-  while ( -1 != (ch_i = getopt_long(argc, argv, "b:n:", lop_u, &lid_i)) ) {
+  u3_Host.dir_c = _main_pier_run(argv[0]);
+
+  while ( -1 != (ch_i=getopt_long(argc, argv, "fn:", lop_u, &lid_i)) ) {
     switch ( ch_i ) {
-      case 'b':
-        u3_Host.ops_u.batch_sz_c = strdup(optarg);
-        break;
-      case 'n':
-        u3_Host.ops_u.til_c = strdup(optarg);
-        break;
-      case '?':
-        fprintf(stderr, "%s\r\n", usage_c);
+      case c3__loom: {
+        c3_w lom_w;
+        c3_o res_o = _main_readw(optarg, u3a_bits + 3, &lom_w);
+        if ( (c3n == res_o) || (lom_w < 20) ) {
+          fprintf(stderr, "error: --loom must be >= 20 and <= %u\r\n", u3a_bits + 2);
+          exit(1);
+        }
+        u3_Host.ops_u.lom_y = lom_w;
+      } break;
```

Comment:
```suggestion
      case c3__loom: {
        if (_main_readw_loom("loom", &u3_Host.ops_u.lom_y)) {
          exit(1);
        }
      } break;
```

Fails on `--loom 33` otherwise.

Reply:
It should use `u3a_bits_max + 1` as the comparison instead. We're already doing this but was probably missed when old changes were ported

Reply:
Thanks, I had just ported the old version straight over.